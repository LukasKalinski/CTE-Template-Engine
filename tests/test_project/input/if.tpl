{* Simple comparation of two variables... *}
{if $foo.bar equals $user}
{/if}

{* Testing with PHP operators... *}
{if $foo.bar == 'foobar!' && $user == $foo.bar}
{/if}

{* Testing:
  - Parentheses
  - Boolean variables / logical not
  - Elseif + Else
*}
{if ($page is equal to 0 or $page is greater than 77) and $isNewYear}
{elseif not $isNewYear}
{else}
{/if}

{* Testing:
  - Function operators
  - Arithmetic expressions
  - Shorter operator aliases (i.e. 'gt' instead of 'is greater than')
*}
{if $page is set and ($page gt 0 or $page is div by 3) and $user not empty}
{elseif $page eq $maxPage-1}
{/if}

{* Testing:
   - Parseable Strings in three combinations
*}
{if $user equals "{$foo.bar} Freeman"}
{elseif $user equals "Freeman {$foo.bar}"}
{elseif $user equals "Mr. {$foo.bar} Freeman"}
{/if}

{* Testing:
   - Parseable Strings with inline arithmetic expression (this is not a good way
     to do it in if tags, used for test purposes only).
*}
{if $page equals "{$maxPage-2}"}
{/if}