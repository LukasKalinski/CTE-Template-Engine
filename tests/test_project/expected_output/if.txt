<?php if($this->v_initial['foo']['bar']==$this->v_initial['user']){?>
<?php }?>

<?php if($this->v_initial['foo']['bar']=='foobar!'&&$this->v_initial['user']==$this->v_initial['foo']['bar']){?>
<?php }?>

<?php if(($this->v_initial['page']==0||$this->v_initial['page']>77)&&$this->v_initial['isNewYear']){?>
<?php }elseif(!$this->v_initial['isNewYear']){?>
<?php }else{?>
<?php }?>

<?php if(isset($this->v_initial['page'])&&($this->v_initial['page']>0||$this->v_initial['page']%3==0)&&!empty($this->v_initial['user'])){?>
<?php }elseif($this->v_initial['page']==$this->v_initial['maxPage']-1){?>
<?php }?>

<?php if($this->v_initial['user']==$this->v_initial['foo']['bar'].' Freeman'){?>
<?php }elseif($this->v_initial['user']=='Freeman '.$this->v_initial['foo']['bar']){?>
<?php }elseif($this->v_initial['user']=='Mr. '.$this->v_initial['foo']['bar'].' Freeman'){?>
<?php }?>

<?php if($this->v_initial['page']==($this->v_initial['maxPage']-2)){?>
<?php }?>